Week2

seungjunGong opened the pull request:
## 📌 PR 요약

🌱 작업한 내용
- Single Activity 적용(메인 화면(home, shorts, live, search, history), My 뷰) 
- Bottom Navigation 구현
- 홈 뷰 구현(Lazy Lists, Horizontal Pager)
- TabLayout 구현
    - 탭 클릭시 카테고리별 데이터 변경
    - 카테고리에서 뒤로가기 시 홈 데이터 변경

🌱 PR 포인트
- 탭구현에서 첫번째 배너를 제외한 아래 두 개의 리스트는 데이터 변경 없이 그대로 사용했습니다..
- TabLayout 구현 시, Material 디자인의 TabLayout을 사용하는 것이 더 나은 선택일지 고민입니다.

## 📸 스크린샷
| 홈화면 | 바텀네비게이션 | 탭레이아웃&MY뷰이동 |
|---------|-------------------|---------------------------|
|<video src="https://github.com/user-attachments/assets/899ffdd6-b4e7-4e86-8754-259ea9a22f03"/>|<video src="https://github.com/user-attachments/assets/eaf2676d-00c0-4a3b-ac50-51d23c1099e7"/>|<video src="https://github.com/user-attachments/assets/407f40d8-3349-4537-9c69-5d8dabd625f0"/>|


## 📮 관련 이슈
- Resolved: #3

seungjunGong committed:
chore: navigation 의존성 추가 #3

- kotlinx-serialization
- compose-navigation

seungjunGong committed:
feat: drawable 아이콘 및 포스터 이미지 추가 #3

seungjunGong committed:
feat: bottom navigation 뷰 연결 #3

seungjunGong committed:
feat: Home Screen 구현 #3

- 뷰 페이저
- lazy row
- 전체 스크롤 구현

seungjunGong committed:
feat: My뷰 연결 #3

seungjunGong committed:
refactor : string 리소스, color theme 적용 #3

seungjunGong committed:
feat: Tablayout 구현 #3

nagaeng commented on app/src/main/java/org/sopt/at/presentation/login/SignInActivity.kt:
SAA 구조니까 우리는 login 관련도 Screen으로만 구현해보는 게 좋을 것 같아요 😎

seungjunGong replied:
네 한개의 액티비티에서 구현해볼게요!

nagaeng commented on app/src/main/java/org/sopt/at/presentation/main/MainActivity.kt:
```diff
@@ -0,0 +1,67 @@
+fun MainActivityPreview() {
+    TvingTheme {
+    }
+}
```
Preview 함수는 빼고 PR 날려주시지요 !! ㅎ_ㅎ

nagaeng commented on app/src/main/java/org/sopt/at/presentation/main/navigation/MainNavigation.kt:
```diff
@@ -0,0 +1,81 @@
+        containerColor = Color.Black,
+    ) {
+        items.forEach { item ->
+            val selected = currentRoute == item.screenRoute::class.qualifiedName
```
여기 잘 동작하나요 ? 

currentRoute는 MainNavigationGraph에서 등록한 home, shorts 같은 route 이름인데 item.screenRoute::class.qualifiedName는 패키지 이름에 클래스 이름 전체 다 넣은 거인데 ,, 타입이 다른데 만약에 잘 돌아간다 하면 비교가 어떻게 되는거지 ...... 무튼

MainNavRoute에 route: String을 명시적으로 추가해주고 (아예 인터페이스에 val route: String 넣고 모든 아이템에 상속 주면 될듯?)
val selected = currentRoute == item.screenRoute.route
이렇게 사용하면 더 깔끔하고 안정적으로 돌아갈 것 같아용

seungjunGong replied:
직렬화를 이용한 방식을 적용하다 보니 처음에 적용이 안되어서 클래스 경로를 받아오게 했네요..
현재 route 값을 문자열로 받으니 route: String 을 이용한 방식이 더 깔끔할거 같네요! 조언 감사합니다:)

nagaeng commented on app/src/main/java/org/sopt/at/presentation/main/screen/MyScreen.kt:
```diff
@@ -28,38 +25,24 @@ import androidx.compose.ui.tooling.preview.Preview
+    paddingValues: PaddingValues,
+    modifier: Modifier = Modifier,
+    id: String = SharedPreferencesManager.getUserId() ?: "",
+    password: String = SharedPreferencesManager.getUserPw() ?: "",
```
password를 파라미터로 받지만, 아무데도 안 쓰고 있는 것 같네용
사용하지 않는 파라미터는 없애는 게 코드가 간결해질 것 같아요 !

nagaeng commented on app/src/main/java/org/sopt/at/presentation/main/screen/home/HomeScreen.kt:
```diff
@@ -0,0 +1,131 @@
+        item {
+            CustomBannerViewPager(
+                items = bannerItems,
+                pagerState = rememberPagerState(pageCount = { bannerItems.size })
```
요 아이템마다 rememberPagerState를 호출하면 recomposition(3주차 세미나 때 얼핏 들었죠?) 때 상태를 잃을 수 있습니다 
val pagerState = rememberPagerState(...)를 아예 HomeScreen 함수 초반부에서 선언하고 CustomBannerViewPager에 넘겨주는 형태도 함 고려해보세요!

nagaeng commented on app/src/main/java/org/sopt/at/presentation/main/screen/home/components/TabBarCategories.kt:
컴포넌트 죽여주게 깔끔하네요
카테고리 쇽샥 가져가버려야겠다 👍🏻

nagaeng commented on app/src/main/java/org/sopt/at/presentation/main/screen/home/data/TabBarCategory.kt:
먼가 지금 뷰모델이랑 data 폴더에 같이 위치 시켜놓은게 딱히 넣을 폴더가 없어서 위치시킨건가요 ?
data는 네트워크 데이터 (DTO, Repository) 같은 외부 데이터 관련 코드를 두는 곳이라서
요 단순 enum이 해당 폴더에 있는게 어색하네요 !!! home 하위에 model 폴더 파서 관리하는 게 좋을 것 같아요

seungjunGong replied:
아하 디렉토리 구조에 대해서 덕분에 공부했네용!!
적용해볼게요

nagaeng commented on app/src/main/res/values/strings.xml:
```diff
@@ -18,6 +23,50 @@
+    <string name="home_tab_title_anime">애니</string>
+    <string name="home_tab_title_news">뉴스</string>
+
+    <!-- 실시간 인기 콘텐츠 제목 -->
```
스트링 추출 너무 잘 했어요 ~~~~ 주석은 빼줍시다 ㅎㅎ

nagaeng reviewed (approved):
심화까지 아주 멋지게 해낸 승준오빠 칭찬합니다 👍🏻👏🏻
스트링 추출부터 리플 효과 제거 같이 전 와비때 안 챙겼던 부분까지 야무지게 챙겼네요 
특히 컴포넌트화가 너무 잘 돼있어서 나중에 쇽샥해가고 싶습니다 ㅎ-ㅎ
너무너무 잘 해서 리뷰 달 것두 없군여. 제가 배워갑니다 고생 많았어 ! 😎

seungjunGong committed:
refactor: SSA 구조 적용(SignIn, SignUp) closes #3

- startDestination 동적 설정
- 직렬화된 route 대신 string 기반 route 비교 적용
- 회원가입 후 로그인 화면에서 Toast 메시지 표시 로직 추가
